### ISSUE TITLE
Remove key generation from tests

### ISSUE THREAD

**Issue by bluejekyll** [bluejekyll/trust-dns#134]
Instead checkin static keys to be used for all testing. This will allow OpenSSL to be dropped as a dependency for tests.

**Comment by bluejekyll** [comment]
This should remove the requirement for OpenSSL as required dev-dependency.

**Comment by shammancer** [comment]
Which tests are using key generation?

**Comment by shammancer** [comment]
Which tests would you want to remove OpenSSL from?

**Comment by shammancer** [comment]
Why do you have so many different tls implementations?

**Comment by shammancer** [comment]
Are the test just testing the same functionality in native-tls, rustls, client/src/tls?

**Comment by bluejekyll** [comment]
There are different TLS implementations for different situations. OpenSSL is the default, most general implementation. Rustls is a pure Rust implementation with a dependency on the *ring* library instead. NativeTLS allows the system default to be used. Each of these is valuable for different scenarios based on peoples needs.

In terms of proof of the DNSSec chain, it's currently split between requiring OpenSSL for some functions and *ring* for others.

In terms of private keys that are generated, and example is on

```rust
let rsa = Rsa::generate(2048).unwrap();
```

on https://github.com/bluejekyll/trust-dns/blob/26ce62f1372d70cd93ff3fdaecc3708da307315f/server/tests/common/authority.rs#L169

If you run `rg '::generate\(' **/*.rs ` at the root of the project, it will give a pretty decent list of what needs to be changed. (this uses ripgrep which can be installed with `cargo install ripgrep` if you're not familiar with it).

What I was thinking of doing for this, was to generate some RSA keys int /tests and use those by loading thing into each test as needed. Does that make sense?

**Comment by briansmith** [comment]
> What I was thinking of doing for this, was to generate some RSA keys int /tests and use those by loading thing into each test as needed. Does that make sense?

Yes, that makes sense. In particular, I recommend generating them in PKCS#8 format because that's the format that every crypto library seems to natively support now.

Uncoincidentally, I wrote a guide to generating PKCS#8-formatted keys in all the necessary formats and key sizes (except legacy RSA key sizes) here: https://gist.github.com/briansmith/2ee42439923d8e65a266994d0f70180b

**Comment by briansmith** [comment]
Also, in case you don't know, you can use `include_bytes!` to include a file as a `&[u8]` value. See the *ring* tests for examples.

**Comment by briansmith** [comment]
Once this is done, all the *ring* tests can be run even when openssl is disabled. That should be done after this.
